src/Test/L0/Worker/TaskManagerL0.cs

```diff
@@ -356,130 +356,37 @@ public async void DownloadsTasks()
-                    }
-                }
-            }
-            finally
-            {
-                Teardown();
-            }
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void PreservesTaskZipTaskWhenAlwaysExtractTask()
+        {
+            PreservesTaskZipTask(true, false);
         }
 
         // TODO: Add test for Extract
         [Fact]
         [Trait("Level", "L0")]
         [Trait("Category", "Worker")]
-        public void ExtractsAnAlreadyDownloadedZipToTheCorrectLocation()
+        public void ExtractsAnAlreadyDownloadedZipToTheCorrectLocationWhenInSignatureVerification()
         {
-            try
-            {
-                // Arrange
-                using (var tokenSource = new CancellationTokenSource())
-                using (var _hc = Setup(tokenSource, signatureVerificationEnabled: true))
-                {
-                    var bingGuid = Guid.NewGuid();
-                    string bingTaskName = "Bing";
-                    string bingVersion = "1.21.2";
-                    var taskStep = new Pipelines.TaskStep
-                    {
-                        Name = bingTaskName,
-                        Reference = new Pipelines.TaskStepDefinitionReference
-                        {
-                            Id = bingGuid,
-                            Name = bingTaskName,
-                            Version = bingVersion
-                        }
-                    };
-                    string zipDestDirectory = Path.Combine(_hc.GetDirectory(WellKnownDirectory.TaskZips), $"{bingTaskName}_{bingGuid}_{bingVersion}.zip");
-                    Directory.CreateDirectory(_hc.GetDirectory(WellKnownDirectory.TaskZips));
-                    // write stream to file
-                    using (Stream zipStream = GetZipStream())
-                    using (var fileStream = new FileStream(zipDestDirectory, FileMode.Create, FileAccess.Write))
-                    {
-                        zipStream.CopyTo(fileStream);
-                    }
-
-                    // Act
-                    _taskManager.Extract(_ec.Object, taskStep);
+            ExtractsAnAlreadyDownloadedZipToTheCorrectLocation(true, false);
+        }
 
-                    // Assert
-                    string destDirectory = Path.Combine(
-                        _hc.GetDirectory(WellKnownDirectory.Tasks),
-                        $"{bingTaskName}_{bingGuid}",
-                        bingVersion);
-                    Assert.True(File.Exists(Path.Combine(destDirectory, Constants.Path.TaskJsonFile)));
-                }
-            }
-            finally
-            {
-                Teardown();
-            }
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void ExtractsAnAlreadyDownloadedZipToTheCorrectLocationWhenExtractTask()
+        {
+            ExtractsAnAlreadyDownloadedZipToTheCorrectLocation(true, false);
```

Comment:
Same comment as above, we have to two tests with the same call?

Reply:
I am guessing this should be switched to `false, true`. This would be helped by adding the named parameters.

Reply:
The true false's are different.  I like them in 2 separate tests cause if 1 fails is super obvious why.  Check out the names.  It's just a line in the test so not expensive to do.